server/boot/01_mongo_id_handler.js

```diff
@@ -2,8 +2,15 @@ module.exports = function(app){
 	var mongoDs = app.datasources.mongoDs;
  	var ObjectID = mongoDs.connector.getDefaultIdType();
  	var RoleMapping = app.models.RoleMapping;
+ 	var orgUser = app.models.orgUser;
+ 	var Role = app.models.Role;
 
  	RoleMapping.defineProperty('principalId', {
  		type: ObjectID,
  	});
+ 	
+ 	RoleMapping.belongsTo(orgUser);
```

Comment:
I think we don't need to add all those lines programatically but we might need this one 
```
orgUser.hasMany(Role, {through: RoleMapping, foreignKey: 'principalId'});
```
to fetch all the roles for a user. As when I tried to include the roles in currentUser middleware, I was getting an error.  And also I was in a different thought process of how to do the role resolver thing I forgot to do more experimentations with this :(